### ISSUE TITLE
BlueHost :: Failed Save Warning :: Can't export the layout if my connection is down

### ISSUE THREAD

**Issue by FaycalBoutam** [elegantthemes/Divi-Beta#949]
### Problem:

So I'm building my page, then I turned off my connection (connection is down), I went to save the page and I got the warning window "Your save has failed" with the option to export the backup.

1 We could add that one of reasons the save is failed is internet connection is down, and the first thing to check is your internet connection before suggesting contacting host... etc.
2 In this case I can not get my back up, clicking the download backup button doesn't do anything, using the VB export page's JSON file is also not working so I'm stuck and I must keep this tab opened until my connection is back again else I would lose my work.

It makes sense that you can't export something from the internet, but is there a backup saved to the browser that can be exported as JSON file?

![1](https://cloud.githubusercontent.com/assets/16414613/22757060/45b46132-ee41-11e6-8630-21fdee26e20f.png)

### Attached PR
* https://github.com/elegantthemes/submodule-builder/issues/1764

**Comment by Jronk** [comment]
There is a backup saved in the browser, so at this point its only purpose/ability is to be used upon a refresh for example in which you would be offered to restore from the backup. The drawback from what the browser backup offers however is that if you close the tab the browser backup will be lost. Its not a real drawback per se, but compared to a downloaded backup, you could hold on to that for as long as needed. Seems like we need to let the browser backup be exported, and able to be re-imported again, so seems like a new version of portability for VB only is needed, which means most likely there is plenty of discussion needed.

**Comment by kary4** [comment]
The same problem here. Also, I've noticed that when you click Try Again button the popup does not appear again. If I understand it should appear again and again when you click it and you did not fix the real problem (for example internet connection).

**Comment by Jronk** [comment]
Fikri was sent to develop the fail modal, and he decided to use portability, with no knowledge of the browser backup being dev’ed at the same time. So, when they both were finished, there resulted two ways of trying to do the same thing. If that fail modal were dev’ed after autosave/backup was finished, backup prob would have been used instead of portability. Without the autosave/backup system existing, his solution of utilizing the portability system would be correct and perfect. But since the autosave/backup system exists, this fail modal needs to be more straightforward:

"Something went wrong/your internet connect is gone… but dont worry, your latest changes are backed up in the browser, and will be autosaved to the server as soon as your connection is restored"

So, in a nutshell, kill the "download backup" aspect of the fail modal

@timayhelen?

**Comment by timayhelen** [comment]
@Jronk I think the ability to download a backup copy is important. It will still be useful for other scenarios.

**Comment by timayhelen** [comment]
@lots0logs The builder always fails to load for me in this branch. I notice the following error:

![screen shot 2017-02-16 at 8 37 08 pm](https://cloud.githubusercontent.com/assets/3333028/23052648/c3f922fc-f487-11e6-9e55-8cc9fe0ab7ab.png)

**Comment by lots0logs** [comment]
@timayhelen Could you try clearing your browser cache? Its working fine for me. I tried it on Firefox (which I hardly ever use) just to be sure.
